src/Sylius/Bundle/ApiBundle/ApiPlatform/Messenger/Processor.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\ApiPlatform\Messenger;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use Sylius\Bundle\ApiBundle\StateProcessor\Command\CommandAwareInputDataProcessorInterface;
+
+final readonly class Processor implements ProcessorInterface
+{
+    public function __construct(
+        private CommandAwareInputDataProcessorInterface $commandDataInputProcessor,
+        private ProcessorInterface $decoratedProcessor,
+    ) {
+    }
+
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): mixed
+    {
+        $data = $this->commandDataInputProcessor->process($data, $operation, $uriVariables, $context);
```

Comment:
I guess it is not a bad idea to move the logic from `CommandAwareInputDataProcessor` here, especially that I don't see any other use for this service, and this way it will be clearer. I needed a moment myself to understand the relationship between the interfaces 😅